flang/include/flang/Frontend/CodeGenOptions.def

```diff
@@ -39,6 +39,7 @@ ENUM_CODEGENOPT(RelocationModel, llvm::Reloc::Model, 3, llvm::Reloc::PIC_) ///<
 ENUM_CODEGENOPT(DebugInfo,  llvm::codegenoptions::DebugInfoKind, 4,  llvm::codegenoptions::NoDebugInfo) ///< Level of debug info to generate
 ENUM_CODEGENOPT(VecLib, llvm::driver::VectorLibrary, 3, llvm::driver::VectorLibrary::NoLibrary) ///< Vector functions library to use
 ENUM_CODEGENOPT(FramePointer, llvm::FramePointerKind, 2, llvm::FramePointerKind::None) ///< Enable the usage of frame pointers
+ENUM_CODEGENOPT(CodeModel, llvm::CodeModel::Model, 5, llvm::CodeModel::Model::Small)
```

Comment:
What was your reasoning for defaulting to `llvm::CodeModel::Model::Small` here? So far as I can tell, clang avoids setting it when the option is not specified explicitly: https://github.com/llvm/llvm-project/blob/44df1167f88cabbb4cfde816f279337379ea30b3/clang/lib/CodeGen/CodeGenModule.cpp#L1304

In LLVM, the code model logic seems to get overridden in each implementation of LLVMTargetMachine. The first one I looked at does default to llvm::CodeModel::Model::Small, but I don't know whether that will always be true for every target.

I think it would be safer to store this as a std::optional instead of a bitfield.

Reply:
On a second look I think you're right. I'm not sure how to represent that in this CodeGenOptions.def file though, or if it isn't possible and I need to do this differently. Perhaps @banach-space can suggest something?

Reply:
I would add it as an `optional` to `CodeGenOptions` in `CodeGenOptions.h`

Reply:
Rightly or wrongly, I've changed it to copy the way that Clang does it, and store the string in CodeGenOptions, then parse it (twice, but sharing the code to do so, rather than implementing the same-ish code twice in Clang).